mne/export/_edf.py

```diff
@@ -261,6 +266,7 @@ def _export_raw(fname, raw, physical_range, add_ch_type):
             onset = onset * 10000
             duration = duration * 10000
             if hdl.writeAnnotation(onset, duration, desc) != 0:
+                hdl.close()
```

Comment:
This solution is less clean / DRY than a solution that uses a context manager like `with hdl:` or, if that's not an option:
```
@contextmanager
def _close_file(fid):
    try:
        yield
    finally:
        fid.close()
```
But we can clean this up in a follow-up PR